### DISCUSSION TITLE
v1-route-convention can't start server problem

### DISCUSSION THREAD

**Discussion by dealtek** [remix-run/remix#6256] [Q&A] [answered]
Hi all,  getting started w remix I tried to set as suggested below:

but now run npm dev has errors:

Error: Cannot add empty string to PrefixLookupTrie
    at PrefixLookupTrie.add (/Users/me/try1/node_modules/@remix-run/dev/dist/config/flat-routes.js:39:23)
    at flatRoutesUniversal (/Users/me/try1/node_modules/@remix-run/dev/dist/config/flat-routes.js:144:18)
    at flatRoutes (/Users/me/try1/node_modules/@remix-run/dev/dist/config/flat-routes.js:100:23)
    at Object.readConfig (/Users/me/try1/node_modules/@remix-run/dev/dist/config.js:210:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.dev (/Users/me/try1/node_modules/@remix-run/dev/dist/cli/commands.js:211:18)
    at async dev (/Users/me/try1/node_modules/@remix-run/dev/dist/cli/run.js:194:3)
    at async Object.run (/Users/me/try1/node_modules/@remix-run/dev/dist/cli/run.js:477:7)

----------

To ensure that you can follow along smoothly, you should therefore switch to the "v1" structure as well - which can be done by following these steps:

1) Execute the below command, inside of the created Remix project folder:

   ` npm i @remix-run/v1-route-convention`

---> also next I replaced previous content with below (not added extra is that correct?) <----

2) Adjust remix.config.js (in the project) to contain this content:

`const {
      createRoutesFromFolders,
    } = require("@remix-run/v1-route-convention");
     
    /** @type {import('@remix-run/dev').AppConfig} */
    module.exports = {
      future: {
        // makes the warning go away in v1.15
        v2_routeConvention: true,
      },
     
      routes(defineRoutes) {
        // uses the v1 convention, works in v1.15+ and v2
        return createRoutesFromFolders(defineRoutes);
      },
    };`

**Comment by machour** [comment]
Hey @dealtek, could you describe how you created and tried to run your project initially? It seems that you're mixing v1 & v2 routes conventions, and that won't work

**Reply by dealtek** [reply]
Hi machour - thanks for your help. I was learning from this course by Maximilian Schwarzmüller

https://www.udemy.com/course/remix-course/
https://github.com/academind/remix-practical-guide-course-resources

I started with :
`npx create-remix@latest`

Then right away he suggests to use (v1):
Adjusting the Created Projects
Remix supports different ways of setting up routes.

There's a so-called "v1" and "v2" structure - though it's important to note that these are just different conventions and no structure is better than the other.

This course was recorded the "v1" structure.

New Remix projects come with the "v2" structure set as a default.

To ensure that you can follow along smoothly, you should therefore switch to the "v1" structure as well - which can be done by following these steps:

those steps I Listed below...

Also his course starts with = index.js and the new download = _index.js so that might be part of the issue?

**Comment by marcfalk** [comment, marked as answer]
Although I am not using `@remix-run/v1-route-convention` I did run into the same error message. This thread was the only result on Google so I thought my solution might help OP or someone swinging by.

TLDR: Make sure `.DS_Store` and other non-route files are not present in the `/routes` directory.

I had this config using Remix v. 1.16.0:

```
module.exports = {
  cacheDirectory: './node_modules/.cache/remix',
  future: {
    v2_errorBoundary: true,
    v2_routeConvention: true,
    v2_normalizeFormMethod: true,
  },
}
```

And I got this error:

```
Error: Cannot add empty string to PrefixLookupTrie
    at PrefixLookupTrie.add (/path_to_project/node_modules/@remix-run/dev/dist/config/flat-routes.js:38:23)
    at flatRoutesUniversal (/path_to_project/node_modules/@remix-run/dev/dist/config/flat-routes.js:141:18)
    at flatRoutes (/path_to_project/node_modules/@remix-run/dev/dist/config/flat-routes.js:97:23)
    at Object.readConfig (/path_to_project/node_modules/@remix-run/dev/dist/config.js:232:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.dev (/path_to_project/node_modules/@remix-run/dev/dist/cli/commands.js:232:18)
    at async Object.run (/path_to_project/node_modules/@remix-run/dev/dist/cli/run.js:505:7)
```

I tried installing [remix-flat-routes](https://github.com/kiliman/remix-flat-routes). According to the docs I used this config:

```
const { flatRoutes } = require('remix-flat-routes')
module.exports = {
  cacheDirectory: './node_modules/.cache/remix',
  ignoredRouteFiles: ['**/*'],
  future: {
    v2_errorBoundary: true,
    v2_normalizeFormMethod: true,
  },
  routes: async (defineRoutes) => {
    return flatRoutes('routes', defineRoutes)
  },
}
```

That gave me another and more explicit error message:

```
Error: Invalid route module file: /path_to_project/app/routes/.DS_Store
    at /path_to_project/node_modules/@remix-run/dev/dist/config/routesConvention.js:73:11
    at visitFiles (/path_to_project/node_modules/@remix-run/dev/dist/config/routesConvention.js:230:7)
    at defineConventionalRoutes (/path_to_project/node_modules/@remix-run/dev/dist/config/routesConvention.js:61:3)
    at Object.readConfig (/path_to_project/node_modules/@remix-run/dev/dist/config.js:232:30)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Object.dev (/path_to_project/node_modules/@remix-run/dev/dist/cli/commands.js:232:18)
    at async Object.run (/path_to_project/node_modules/@remix-run/dev/dist/cli/run.js:505:7)
```

I cleared my whole project of `.DS_Store`-files by running the following command in my project src folder:

`find ./ -name ".DS_Store" -print -delete`

Now, I was able to run `npm run dev` successfully with and without `remix-flat-routes`.

✌️

**Reply by asciant** [reply]
This was exactly my issue, thank you!

**Reply by kiliman** [reply]
The issue here is that when you want to override the default convention, you need to ignore **all** files in the main config `ignoredRouteFiles: ['**/*']`. This way, the new convention can handle the files. However, now it's the *plugin's* responsibility to ignore non-route files like `.DS_Store`. Both the `v1-route-convention` and `remix-flat-routes` have an `ignoredRouteFiles` option that you can pass `['**/.*']`.

**Reply by zachdtaylor** [reply]
Thanks for posting this. Saved me a lot of time!

**Reply by beerlington** [reply]
I ran into this too and added the following to my project's `.gitignore` file to make it easier to find these files if they appear again:

```
!/app/routes/**/.DS_Store
```

**Reply by cyberwombat** [reply]
Adding a note that its not just the routes folder. I had no .DS_Store in there but I had some in other folders including `public`. Clearing the entire project was the solution. Thanks!

**Reply by Otann** [reply]
You saved my day, thank you!

**Comment by dealtek** [comment]
Hi [marcfalk](https://github.com/marcfalk) - wow that was some good research I would not know about the `.DS_Store` files - Thanks
